pkgs/build-support/cc-wrapper/ld-wrapper.sh

```diff
@@ -135,10 +141,10 @@ if [ "$NIX_DONT_SET_RPATH" != 1 ]; then
     # so, add the directory to the rpath.
     # It's important to add the rpath in the order of -L..., so
     # the link time chosen objects will be those of runtime linking.
-    for i in ${libDirs[@]}; do
-        for j in ${libs[@]}; do
-            if [ -f "$i/lib$j.so" ]; then
-                addToRPath $i
+    for dir in "${libDirs[@]}"; do
+        for file in "$dir"/lib*.so; do
+            if [ "${libs[${file##*/}]}" ]; then
```

Comment:
If you meant that the whole conditional `if [ "${libs[${file##*/}]}" ]` should be removed, it is not the case because the main purpose of this loop is to filter out directories with libraries none of which are needed.